functions/go/set-namespace/main.go

```diff
@@ -33,11 +38,31 @@ func main() {
 					yaml.FieldSetter{StringValue: namespace}); err != nil {
 					return err
 				}
+			} else {
+				if !allowClusterScoped {
+					apiVersionNode, err := node.Pipe(yaml.Lookup("apiVersion"))
+					if err != nil {
+						continue
+					}
+					apiVersion := yaml.GetValue(apiVersionNode)
+					nameNode, err := node.Pipe(yaml.Lookup("metadata", "name"))
+					if err != nil {
+						continue
+					}
+					name := yaml.GetValue(nameNode)
+					resID := fmt.Sprintf("apiVersion: %s, kind: %s, name: %s", apiVersion, kind, name)
+					clusterScopedResources = append(clusterScopedResources, resID)
+				}
 			}
 		}
+		if len(clusterScopedResources) > 0 {
```

Comment:
what's the plan for cluster-scoped custom resources not hardcoded in implementation of isNamespaceable()?

Reply:
For these cluster-scoped custom resources, ACM will fail to apply the resources for the following error:
```
KNV1052: cluster-scoped resources MUST NOT declare metadata.namespace
```

Reply:
After https://github.com/kubernetes-sigs/kustomize/pull/2620 is merged, we can also use the annotation to avoid adding non-application k8s resources (e.g., functions, constraints) into the hydrated config.

Reply:
If ACM somehow does not cover all the cluster-scoped resource kinds, and the attempts to apply the cluster-scoped resources into a given namespace most likely will fail due to lack of permission.

Reply:
This check here is best-effort. There will be many cluster-scoped CRDs that are not hardcoded in kyaml, and won't be detected. We should be clear about that and document the UX to the user.

Reply:
I clarified this in the command flags.